pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -563,3 +567,64 @@ func (d *namespacedResourcesDeleter) estimateGracefulTerminationForPods(ns strin
 	}
 	return estimate, nil
 }
+
+// updateStatusCondition does best effort to update status condition for
+// a namespace if namespace deletion didn't finish. In case of a failure, it
+// does not propagate its error further, instead it logs it and the error
+// from namespace deletion is propagated.
+func (d *namespacedResourcesDeleter) updateStatusCondition(nsName string, deletionErr error) {
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		ns, err := d.nsClient.Get(nsName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		changeStatusConditionWithErr(ns, deletionErr)
+		_, err = d.nsClient.UpdateStatus(ns)
```

Comment:
this updates always due to the probe time. This deserves a comment here why we don't call DeepEqual on old and new status.

Reply:
thanks, I absolutely missed that. I would like to use your comment and check status update before sending to the API server